zingolib/src/wallet/notes/interface.rs

```diff
@@ -0,0 +1,70 @@
+use incrementalmerkletree::{Hashable, Position};
+use zcash_primitives::{memo::Memo, merkle_tree::HashSer, transaction::TxId};
+
+use super::super::{
+    data::TransactionRecord,
+    keys::unified::WalletCapability,
+    traits::{FromBytes, FromCommitment, Nullifier, ReadableWriteable, ToBytes},
+    Pool,
+};
+
+pub trait NoteInterface: Sized {
+    fn spent(&self) -> &Option<(TxId, u32)>;
+    fn is_spent(&self) -> bool {
+        Self::spent(self).is_some()
+    }
```

Comment:
it makes logical sense to me to have is_spent as the first or last fn so spent and spent_mut etc. are grouped together. just an organisation thing